### PR TITLE
Somewhere

### PR DESCRIPTION
**Pull request by dinosaure**

### CODE REVIEW

File: lib/field.mli [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  | Mailboxes : Mailbox.t list v
+  | Mailbox : Mailbox.t v
+  | Addresses : Address.t list v
+  | MessageID : Rfc822.nonsense Rfc822.msg_id v
+  | Phrase_or_message_id : phrase_or_message_id list v
+  | Unstructured : Unstructured.t v
+  | Phrases : phrase list v
+  | Content : Content.t v
+  | Resent : Resent.t v
+  | Trace : Trace.t v
```
Comment (hannesm) [inline lib/field.mli]: sorry for jumping in here, but I've a question: what is the relationship between `t` and `v`? both seem to be GADTs that carry name-value relationships..

Comment (dinosaure) [reply lib/field.mli]: `'a t` is a field-name and `'a v` is type of values possible. I'm not clear about the design on them but, in my case, multiple field names can have the same kind of value (like `Cc` and `Bcc`). Then, an approach available in `Hd` module is to describe what we expect like: `Field "X-Received"` with `Trace` kind of value.

Comment (hannesm) [reply lib/field.mli]: ah, thanks for the quick reply!

Comment (dinosaure) [reply lib/field.mli]: [This type](https://github.com/mirage/mrmime/blob/master/lib/hd.mli#L14) should be equal to `'a Field.v`.

**Pull request merged**